src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useDispatch, useSelector } from 'react-redux';
+import { deleteCurrentTodo, selectedTodo } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const todoModal = useSelector(selectedTodo);
+  const dispatch = useDispatch();
+
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoadingModal, setIsLoadingModal] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsLoadingModal(true);
+    getUser(todoModal!.userId)
```

Comment:
The use of the non-null assertion operator (`!`) on `todoModal` assumes that `todoModal` is always defined. This could lead to runtime errors if `todoModal` is null. Consider adding a check to ensure `todoModal` is not null before attempting to access `todoModal.userId`.